### ISSUE TITLE
Support multiple filters in aggregations

### ISSUE THREAD

**Issue by sezaru** [ash-project/ash#1265]
**Is your feature request related to a problem? Please describe.**
Right now only read actions support multiple filters calls at the same time, having the same behavior in aggregations would consolidate the behavior and make things simpler.

Right now, if you try to do that you will get the following error:

> Multiple values for key `:filter`

**Describe the solution you'd like**

Allow users to write aggregations like this one:

```elixir
    count :total_students_this_month, :students do
      filter expr(inserted_at >= fragment("date_trunc('month', now())"))
      filter expr(inserted_at < fragment("date_trunc('month', now()) + interval '1 month'"))
    end
```

**Comment by TravelCurry02** [comment]
From what I gather, read actions support multiple filter lines (combined with and), but aggregates only support a single filter, which is why trying to add more than one filter in an aggregate leads to the “Multiple values for key :filter” error.

**Comment by zachdaniel** [comment]
Yes, that is correct. We could almost certainly implement the same kind of logic that read actions have for aggregates 😄

**Comment by yenrab** [comment]
Looks like a fix was committed but this wasn't closed?
